src/JuliusSweetland.OptiKey.UnitTests/Extensions/PointExtensionsTests.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System.Collections.Generic;
+using System.Windows;
+using JuliusSweetland.OptiKey.Extensions;
+using NUnit.Framework;
+
+namespace JuliusSweetland.OptiKey.UnitTests.Extensions
+{
+    [TestFixture]
+    public class PointExtensionsTests
+    {
+        [Test]
+        public void TestCalculateCentrePoint()
```

Comment:
@SamPlusPlus That's interesting. Rounding to the nearest even number is pretty arbitrary, but I'm not particularly concerned about it as the decision to round in either direction would also be arbitrary. From a practical perspective we're dealing in pixels, so half a pixel is meaningless, and choosing one pixel to the left, right, up, or down shouldn't make much difference. My only concern with explicitly performing a Math.Ceiling would be that the result still needs to be converted to an Int.

If you would like to change this to a Math.Ceiling and then convert to an Int then I'll happily accept the pull request.